protocol/metrics/consumer_optimizer_qos_client.go

```diff
@@ -68,57 +75,56 @@ func NewConsumerOptimizerQoSClient(endpointAddress string, interval ...time.Dura
 	return &ConsumerOptimizerQoSClient{
 		consumerOrigin:                     hostname,
 		queueSender:                        NewQueueSender(endpointAddress, "ConsumerOptimizerQoS", nil, interval...),
-		optimizers:                         &common.SafeSyncMap[string, OptimizerInf]{},
-		providerToChainIdToRelaysCount:     &common.SafeSyncMap[string, *common.SafeSyncMap[string, *atomic.Uint64]]{},
-		providerToChainIdToNodeErrorsCount: &common.SafeSyncMap[string, *common.SafeSyncMap[string, *atomic.Uint64]]{},
-		providerToChainIdToEpochToStake:    &common.SafeSyncMap[string, *common.SafeSyncMap[string, *common.SafeSyncMap[uint64, uint64]]]{},
+		optimizers:                         map[string]OptimizerInf{},
+		chainIdToProviderToRelaysCount:     map[string]map[string]uint64{},
```

Comment:
careful please verify the following: 

when you have a map[string]reference VS map[string]pointer there is a huge difference when working with the keys.

for example:

value, ok := map["bobo"] 
if value is not a pointer but a reference, changes on value wont affect the map. 

can you verify that all writes are pointing directly to 
map["bobo"] = X 
and not 
value = X 
?

Reply:
Talked about this offline, resolving